x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/spatial/140_synthetic_source.yml

```diff
@@ -411,6 +411,85 @@
   - match: { _source.point.lon: -71.34000029414892 }
   - match: { _source.point.lat: 41.119999922811985 }
 
+---
+"geo_point with ignore_malformed":
+  - requires:
+      cluster_features: ["mapper.track_ignored_source"]
+      reason: introduced in 8.15.0
+
+  - do:
+      indices.create:
+        index: test
+        body:
+          mappings:
+            _source:
+              mode: synthetic
+            properties:
+              geo_point:
+                type: geo_point
+                ignore_malformed: true
+
+  - do:
+      index:
+        index: test
+        id: "1"
+        body:
+          geo_point:
+            - string: "string"
+              array: [{ "a": 1 }, { "b": 2 }]
+              object: { "foo": "bar" }
+            - lat: 41.12
+              lon: -71.34
+
+  - do:
+      index:
+        index: test
+        id: "2"
+        body:
+          geo_point: ["POINT (-71.34 41.12)", "potato", "POINT (-77.03653 38.897676)"]
+
+  - do:
+      index:
+        index: test
+        id: "3"
+        body:
+          geo_point: ["POINT (-77.03653 1000)", "POINT (-71.34 41.12)"]
+
+  - do:
+      indices.refresh: {}
+
+  - do:
+      get:
+        index: test
+        id: "1"
+
+  - match: { _source.geo_point.0.lon: -71.34000004269183 }
+  - match: { _source.geo_point.0.lat: 41.1199999647215 }
+  - match: { _source.geo_point.1.string: "string" }
+  - match: { _source.geo_point.1.array: [{ "a": 1 }, { "b": 2 }] }
+  - match: { _source.geo_point.1.object: { "foo": "bar" } }
+
+  - do:
+      get:
+        index: test
+        id: "2"
+
+  - match: { _source.geo_point.0.lon: -77.0365300308913 }
+  - match: { _source.geo_point.0.lat: 38.89767596963793 }
+  - match: { _source.geo_point.1.lon: -71.34000004269183 }
+  - match: { _source.geo_point.1.lat: 41.1199999647215 }
+  - match: { _source.geo_point.2: "potato" }
+
+  - do:
+      get:
+        index: test
+        id: "3"
+
+  - match: { _source.geo_point.0.lon: -71.34000004269183 }
+  - match: { _source.geo_point.0.lat: 41.1199999647215 }
+  - match: { _source.geo_point.1: "POINT (-77.03653 1000)" }
+
+
```

Comment:
Should we test also the case where a geo point is empty? Or were only `lat` or `lon` are available? I would expect this to be reflected when reconstructing the document.

Reply:
We test empty values in unit tests. I'll add a test for just one coordinate.